[RM Ch08] Additions of O-RAN alignment (#3345)
anuket-project/anuket-specifications: doc/ref_model/chapters/chapter08.rst

+--------+--------+--------+--------+--------+--------+--------+--------+-------+-------+-------+-------+-------+------+

**Table 8-6:** Comparison of Deployment Topologies

O-RAN alignment and interaction
-------------------------------

O-RAN is an operator alliance group with members from the major Telco Operators, Vendors and other interested eco system participants around the specification of Open Radio Access Networks.  Its task is to cloudify the 3GPP specified RAN Network Functionalities with multi-vendor open interfaces in between the Network Functions, the Cloud Infrastructure and the management them. The Service Management and Orchestration (SMO) of multiple O-Clouds is also during specification including a framework for 3:rd party applications (rApps). There are also a few other open interfaces that are aimed to be specified e.g. for Radio Layer 1 HW Accelerators and some low level Radio functions. 

> On a high level O-RAN covers similar specification grounds as what Anuket do, but there are some noteworthy differences both on specification level and on the aim for how O-Clouds are realized. O-RAN specifies how management and orchestration of the Network Functions and Cloud Infrastructure shall be done with a set of internal Services that also have a set of interface specifications for how the rApps could enhance the management functionality. O-RAN have also articulated that O-Clouds can be distributed over multiple Cloud Sites that are stitched over an externally specified WAN interconnect transport that is not part of the O-RAN.

Reviewer: ```suggestion
On a high level, O-RAN covers similar specification grounds as what Anuket do, but there are some noteworthy differences both on specification level and on the aim for how O-Clouds are realized. O-RAN specifies how management and orchestration of the Network Functions and Cloud Infrastructure shall be done with a set of internal Services that also have a set of interface specifications for how the rApps could enhance the management functionality. O-RAN have also articulated that O-Clouds can be distributed over multiple Cloud Sites connected over an externally-specified WAN interconnect transport that is not part of the O-RAN.
```

Author: Agree on the M call not to change the original text.

.. figure:: ../figures/RM-Ch08-O-RAN_mappedon_Anuket-Image-1.png
   :name: O-RAN architecture mapped onto Anuket Reference Model
   :alt: O-RAN architecture mapped onto Anuket Reference Model

   O-RAN architecture mapped onto Anuket Reference Model

> O-Clouds are in some ways similar to the Anuket Cloud Infrastructure with the notable differences that they have an O-RAN specified interface of how the O-Cloud infrastructure is managed (O2ims) and how workloads (e.g. whole or parts of Network Functions) are deployed on the O-Cloud clusters (O2dms). On a more detailed level the O-Clouds are internally very Layer 2 (Ethernet) centric today with strict requirements of determinism and low latency for Cloud Site internal connectivity in between the Network Functions. The O-Cloud also have the set of O-RAN specified HW Accelerators and an Acceleration Adaptation Layer (AAL) of how they are used from the Network Functions for their Radio-near functions.

Reviewer: ```suggestion
O-Clouds are in some ways similar to the Anuket Cloud Infrastructure with the notable differences that they have an O-RAN specified interface of how the O-Cloud infrastructure is managed (O2ims) and how workloads (e.g. whole or parts of Network Functions) are deployed on the O-Cloud clusters (O2dms). On a more detailed level the O-Clouds are internally very Layer 2 (Ethernet) centric, today with strict requirements of determinism and low latency for Cloud Site internal connectivity in between the Network Functions. The O-Clouds also have the set of O-RAN specified HW Accelerators and an Acceleration Adaptation Layer (AAL) of how they are used from the Network Functions for their Radio-near functions.
```

> A potential alignment between Anuket and O-RAN that could be investigated is to see if Anuket could serve as an under-cloud to the O-Cloud. This would require Anuket to support the O-RAN HW Accelerators and a Layer 2 centric networking infrastructure. It would enable the operators to have an internal Telco Cloud that supports both Core and RAN Network Functions and in the extension possibly also other workloads in a shared Cloud that supports required Telco features and characteristics.

Reviewer: ```suggestion
A potential alignment between Anuket and O-RAN that could be investigated is to see if Anuket could serve as an under-cloud to the O-Cloud. This would require Anuket to support the O-RAN HW Accelerators (like started in RM Ch3 [Example of O-RAN AAL Interface](https://github.com/anuket-project/anuket-specifications/blob/master/doc/ref_model/chapters/chapter03.rst#example-of-o-ran-acceleration-abstraction-layer-interface) and a Layer 2-centric networking infrastructure. It would enable the operators to have an internal Telco Cloud that supports both Core and RAN Network Functions, and in the extension possibly also other workloads in a shared Cloud that supports required Telco features and characteristics.
```

Author: Implemented the link to chapter 3 after a small modification of the words suggested by petorre.
(like started -> (as stated
Layer 2-centric -> Layer2-centric

> A potential alignment between Anuket and O-RAN that could be investigated is to see if Anuket could serve as an under-cloud to the O-Cloud. This would require Anuket to support the O-RAN HW Accelerators (as stated in RM Ch3 section Example of O-RAN AAL Interface :ref:`chapters/chapter03.rs:example of o-ran acceleration abstraction layer interface` and a Layer2-centric networking infrastructure. It would enable the operators to have an internal Telco Cloud that supports both Core and RAN Network Functions, and in the extension possibly also other workloads in a shared Cloud that supports required Telco features and characteristics.

Reviewer 2: To support AAL in Anuket there should be an open source implementation of it.

> A potential alignment between Anuket and O-RAN that could be investigated is to see if Anuket could serve as an under-cloud to the O-Cloud. This would require Anuket to support the O-RAN HW Accelerators (as stated in RM Ch3 [Example of O-RAN AAL Interface](https://github.com/anuket-project/anuket-specifications/blob/master/doc/ref_model/chapters/chapter03.rst#example-of-o-ran-acceleration-abstraction-layer-interface) and a Layer2-centric networking infrastructure. It would enable the operators to have an internal Telco Cloud that supports both Core and RAN Network Functions, and in the extension possibly also other workloads in a shared Cloud that supports required Telco features and characteristics.

Reviewer 2: ```suggestion
A potential alignment between Anuket and O-RAN that could be investigated is to see if Anuket could serve as an under-cloud to the O-Cloud. This would require Anuket to support the O-RAN HW Accelerators (as stated in RM Ch3 section Example of O-RAN AAL Interface :ref:`chapters/chapter03.rs:example of o-ran acceleration abstraction layer interface` and a Layer2-centric networking infrastructure. It would enable the operators to have an internal Telco Cloud that supports both Core and RAN Network Functions, and in the extension possibly also other workloads in a shared Cloud that supports required Telco features and characteristics.
```

   :alt: O-RAN architecture mapped onto Anuket Reference Model

   O-RAN architecture mapped onto Anuket Reference Model

O-Clouds are in some ways similar to the Anuket Cloud Infrastructure with the notable differences that they have an
O-RAN specified interface of how the O-Cloud infrastructure is managed (O2ims) and how workloads (e.g. whole or parts of
Network Functions) are deployed on the O-Cloud clusters (O2dms). On a more detailed level the O-Clouds are internally
very Layer2 (Ethernet) centric, today with strict requirements of determinism and low latency for Cloud Site internal
connectivity in between the Network Functions. The O-Clouds also have the set of O-RAN specified HW Accelerators and an
Acceleration Adaptation Layer (AAL) of how they are used from the Network Functions for their Radio-near functions.

A potential alignment between Anuket and O-RAN that could be investigated is to see if Anuket could serve as an
under-cloud to the O-Cloud. This would require Anuket to support the O-RAN HW Accelerators (as stated in RM Ch3 section
Example of O-RAN AAL Interface :ref:`chapters/chapter03:example of o-ran acceleration abstraction layer interface`

> and a Layer2-centric networking infrastructure. It would enable the operators to have an internal Telco Cloud that

Reviewer 2: ```suggestion
and a Layer2-centric networking infrastructure, what assumes an open source solution for AAL. It would enable the operators to have an internal Telco Cloud that
```

Author: The AAL is more of a specification and interface framework that enable the HA Acceleration vendors (and their potential 3rd party Acceleration function vendors) to adapt the O-Cloud to offer the HW Accelerators and their functions in towards the O-Cloud offered Clusters (e.g. K8s) and also managed the HW Accelerators (in a vendor proprietary way) through HW Acceleration Managers (HAM) with O-RAN standardized functions. 
I don't see that there will be any Open Source implementations of O-Clouds, HAMs or AALs, but they are all part of the O-Cloud box, so maybe I should slightly redraw the picture and have AAL inside the O-Cloud box.

Reviewer 2: Anuket RA2 should be built from open source components, therefore if AAL support is needed in RA2 that would require an open source implementation. Wouldn't the built in mechanisms of Kubernetes, like Device Plugins serve as the HW Acceleration Managers in this architecture?
I would rephrase this sentence in a way then to reflect that O-RAN the alignment should be bidirectional in a way that O-RAN should build on the already available open source technology and RA2 should support the requirements from O-RAN.
